src/components/Project/ProjectView.js

```diff
@@ -46,6 +46,10 @@ const ProjectView = React.forwardRef(
     },
     ref
   ) => {
+    const registerArtifactLink = `/projects/${match.params.projectName}/${
+      artifactKind === 'dataset' ? 'feature-store/datasets' : `${artifactKind}s`
+    }`
```

Comment:
I think this has a bit of "magic" to it.
It "just so happens" that the URL path to redirect to has the artifact kind in it.
Let's try to be more explicit:
- If it's a model, go to URL1
- Else, if it's a dataset, go to URL2
- Else, if it's a file, go to URL3
- Else, go to URL4

For example:
```suggestion
    const registerArtifactLink = `/projects/${match.params.projectName}/${
      artifactKind === 'model'   ? 'models'                 :
      artifactKind === 'dataset' ? 'feature-store/datasets' :
      artifactKind === 'file'    ? 'files'                  :
                                   'artifacts'
    }`
```
(of corse Prettier would change the formatting, but the concept is what matters)